tofhir-engine/src/main/scala/io/tofhir/engine/data/read/FileDataSourceReader.scala

```diff
@@ -31,7 +31,15 @@ class FileDataSourceReader(spark: SparkSession) extends BaseDataSourceReader[Fil
    * @throws NotImplementedError      If the specified source format is not implemented.
    */
   override def read(mappingSource: FileSystemSource, sourceSettings:FileSystemSourceSettings, schema: Option[StructType], timeRange: Option[(LocalDateTime, LocalDateTime)], limit: Option[Int] = Option.empty,jobId: Option[String] = Option.empty): DataFrame = {
-    val finalPath = FileUtils.getPath(sourceSettings.dataFolderPath, mappingSource.path).toAbsolutePath.toString
+
+    // Do not add absolute path if it is a hadoop path
+    val finalPath = if (sourceSettings.dataFolderPath.startsWith("hdfs://")) {
```

Comment:
Does HDFS have other attributes which may be included as config parameters in DataSourceSettings? If so, it will be better to implement a dedicated class for HDFS and implement its specific reader instead of this "if" statement.

Reply:
HDFS has no other attributes we can pass using config parameters in DataSourceSettings.